openssl-dynamic/src/main/c/jnilib.c

```diff
@@ -369,41 +389,49 @@ static jint netty_internal_tcnative_Library_JNI_OnLoad(JNIEnv* env, const char*
 
     return TCN_JNI_VERSION;
 error:
+    if (tcn_global_pool != NULL) {
+        TCN_UNLOAD_CLASS(env, jString_class);
+        apr_terminate();
```

Comment:
consider setting `tcn_global_pool = NULL` and `jString_class = NULL`. this will clarify lifetime expectations for other methods and we will get a null pointer de-reference if other methods try to use them. `netty_internal_tcnative_Library_JNI_OnUnLoad` doesn't do this bcz it is assume to be the last method called in this module, however it could also do the same.

Reply:
Let me change the `TCN_UNLOAD_CLASS ` macro to also set it to `NULL`. This will ensure we do it everywhere for stuff that we unload.